integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/QueryStringPredicateSpecificsIT.java

```diff
@@ -0,0 +1,275 @@
+				.where( f -> f.queryString()
+						.field( absoluteFieldPath )
+						.matching( phrase )
+						.minimumShouldMatchPercent( min ) )
+				.toQuery();
+		if ( docId == null ) {
+			assertThatQuery( query )
+					.hasNoHits();
+		}
+		else {
+			assertThatQuery( query )
+					.hasDocRefHitsAnyOrder( index.typeName(), docId, otherDocIds );
+		}
+	}
+
+	public static List<? extends Arguments> minimumShouldMatchPercent() {
+		// "Here I was, feeding my panda, and the crowd had no word."
+		return List.of(
+				Arguments.of( "feeding my panda", 34, DOCUMENT_1, new String[] { DOCUMENT_3 } ),
+				Arguments.of( "feeding my panda", 67, DOCUMENT_1, new String[] { } ),
+				Arguments.of( "feeding my panda", 100, DOCUMENT_1, new String[] { } ),
+				Arguments.of( "feeding my panda crowd had", 100, DOCUMENT_1, new String[] { } ),
+				Arguments.of( "some other panda less matches", 100, null, new String[] { } )
+		);
+	}
+
+	public static List<? extends Arguments> minimumShouldMatch() {
+		// "Here I was, feeding my panda, and the crowd had no word."
+		return List.of(
+				Arguments.of( "feeding my panda", 1, DOCUMENT_1, new String[] { DOCUMENT_3 } ),
+				Arguments.of( "feeding my panda", 2, DOCUMENT_1, new String[] { } ),
+				Arguments.of( "feeding my panda", 3, DOCUMENT_1, new String[] { } ),
+				Arguments.of( "feeding my panda crowd had", 5, DOCUMENT_1, new String[] { } ),
+				Arguments.of( "some other panda less matches", 5, null, new String[] { } )
+		);
+	}
+
+	public static List<? extends Arguments> allowLeadingWildcard() {
+		// "An elephant ran past John."
+		return List.of(
+				Arguments.of( "?lephant", null, DOCUMENT_4 ),
+				Arguments.of( "?lephant", true, DOCUMENT_4 ),
+				Arguments.of( "?lephant", false, null ),
+
+				Arguments.of( "*hant", null, DOCUMENT_4 ),
+				Arguments.of( "*hant", true, DOCUMENT_4 ),
+				Arguments.of( "*hant", false, null ),
+
+				Arguments.of( "ele?hant", null, DOCUMENT_4 ),
+				Arguments.of( "ele?hant", true, DOCUMENT_4 ),
+				Arguments.of( "ele?hant", false, DOCUMENT_4 ),
+
+				Arguments.of( "el*nt", null, DOCUMENT_4 ),
+				Arguments.of( "el*nt", true, DOCUMENT_4 ),
+				Arguments.of( "el*nt", false, DOCUMENT_4 )
+		);
+	}
+
+	public static List<? extends Arguments> phraseSlop() {
+		// Here I was, feeding my panda, and the crowd had no word
+		String phraseReorderedWords = "\"had crowd\"";
+		String phraseWithoutTwoWordsInTheMiddle = "\"crowd word\"";
+
+		return List.of(
+				// default value means phrases should be exact as in text:
+				Arguments.of( phraseReorderedWords, 0, null ),
+				// Transposed terms have a slop of 2
+				Arguments.of( phraseReorderedWords, 2, DOCUMENT_1 ),
+				// not enough so should not match anything
+				Arguments.of( phraseReorderedWords, 1, null ),
+				Arguments.of( phraseWithoutTwoWordsInTheMiddle, 1, null ),
+				Arguments.of( phraseWithoutTwoWordsInTheMiddle, 2, DOCUMENT_1 ),
+				Arguments.of( phraseWithoutTwoWordsInTheMiddle, 5, DOCUMENT_1 )
+		);
+	}
```

Comment:
This would probably be easier to read if each "parameter method" was just after the corresponding test method... no?

Reply:
good point 😃 will update it